src/System.Management.Automation/engine/remoting/commands/PushRunspaceCommand.cs

```diff
@@ -1272,10 +1284,16 @@ private RemoteRunspace GetRunspaceForSSHSession()
         {
             var sshConnectionInfo = new SSHConnectionInfo(this.UserName, ResolveComputerName(HostName), this.KeyFilePath, this.Port);
             var typeTable = TypeTable.LoadDefaultTypeFiles();
-            var remoteRunspace = RunspaceFactory.CreateRunspace(sshConnectionInfo, this.Host, typeTable) as RemoteRunspace;
-            remoteRunspace.Open();
-            remoteRunspace.ShouldCloseOnPop = true;
 
+            // Use the class _tempRunspace field while the runspace is being opened so that StopProcessing can be handled at that time.
+            // This is only needed for SSH sessions where a Ctrl+C during an SSH password prompt can abort the session before a connection
+            // is established.
+            _tempRunspace = RunspaceFactory.CreateRunspace(sshConnectionInfo, this.Host, typeTable) as RemoteRunspace;
+            _tempRunspace.Open();
+            _tempRunspace.ShouldCloseOnPop = true;
```

Comment:
Isn't this an order of operations issue? It seems like the flag should be set before the call to Open().

Reply:
Do you mean the ShouldCloseOnPop property?  No.  A runspace pop cannot occur before this method completes.